### PR TITLE
Frontend: Search, modify & add property managers

### PR DESCRIPTION
**Pull request by annisassi**
# Hitas Pull Request

# Description

New features:
- Search, modify and add property managers in the codes view
- Modify property manager in the apartment details view
Modification:
- "Unmodified" user info to owner modification modal

## Pull request checklist

Check the boxes for each DoD item that has been completed:

- **Testing**
  - [x] Changes have been tested
  - [ ] Automatic tests have been added
- **Database**
  - [ ] Database migrations will work in the DEV & TEST environments
  - [ ] initial.json has been updated to work with migrations
  - [ ] Oracle migration has been updated
- **Documentation**
  - [ ] Tooltips have been added in the frontend for all new fields
  - [ ] OpenAPI definitions have been updated
  - [ ] Test instructions have been written for the customer in the appropriate ticket in Jira
  - [ ] Terminology page in Confluence has been updated

## Test plan

- Modify the property manager info in the apartment details view and check that the validations work as expected and the data updates after modification. 
In the property managers tab in codes view: 
- Search property managers by name and/or email and check you can modify the selected property manager and the data updates after modification.
- Add new property manager and check it is added in the property managers list

## Tickets

This pull request resolves all or part of the following ticket(s): HT-640, HT-644

### CODE REVIEW

**Review by ranta — CHANGES_REQUESTED** [review]

File: frontend/src/styles/components/_Codes.sass [hunk 1]
```diff
@@ -102,6 +102,10 @@
...
-      margin-left: auto
-      margin-right: auto
-      margin-top: 1em
+    @include flexbox()
+    @include flex-flow(column nowrap)
+    @include align-items(center)
+    margin-bottom: $spacing-layout-xs
+
+  .list-footer-item
+    margin-top: $spacing-layout-xs
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/styles/components/_Codes.sass]: There is a contradiction in the prettier rules as it did not accept the final line break in .sass files (accepts everywhere else), so I had to remove this rule from my IDE.

File: frontend/src/app/services.ts [hunk 2]
```diff
@@ -179,7 +180,7 @@ export const hitasApi = createApi({
         },
         ...(!Config.token && {credentials: "include"}),
     }),
-    tagTypes: ["HousingCompany", "Apartment", "Index", "Owner", "ExternalSaleData", "ThirtyYearRegulation"],
+    tagTypes: ["HousingCompany", "Apartment", "Index", "Owner", "Manager", "ExternalSaleData", "ThirtyYearRegulation"],
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/app/services.ts]: This term was uniform with the "owner" term which is not "PropertyOwner" ;D  Are there any other managers in this context? I changed it anyway...

File: frontend/src/app/services.ts [hunk 3]
```diff
@@ -283,7 +285,7 @@ const detailApi = hitasApi.injectEndpoints({
     endpoints: (builder) => ({
         getHousingCompanyDetail: builder.query<IHousingCompanyDetails, string>({
             query: (id) => `housing-companies/${id}`,
-            providesTags: (result, error, arg) => [{type: "HousingCompany", id: arg}],
+            providesTags: (result, error, arg) => [{type: "HousingCompany", id: arg}, {type: "Manager"}],
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/app/services.ts]: Added it to PropertyManager tags.

File: frontend/src/common/components/ModifyManagerModal.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                    name="email"
+                    label="Sähköpostiosoite"
+                    formObject={managerFormObject}
+                />
+                {
+                    // show info about the disabled saving of the unmodified form
+                    !managerFormObject.formState.isDirty && (
+                        <p className="error-message">Lomakkeen tietoja ei ole muutettu</p>
+                    )
+                }
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/common/components/ModifyManagerModal.tsx]: The latter is a good idea. I changed the save button to close the mdoal if no changes are made, as it confuses the user if there is no visible error and still the save button is disabled.

File: frontend/src/common/components/ModifyManagerModal.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                    <SaveButton
+                        isLoading={isSaveManagerLoading}
+                        type="submit"
+                        buttonText="Tallenna"
+                        disabled={!managerFormObject.formState.isDirty || !managerFormObject.formState.isValid}
+                    />
+                </div>
+            </form>
+
+            <Dialog.ActionButtons />
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/common/components/ModifyManagerModal.tsx]: Removed

File: frontend/src/features/apartment/ApartmentDetailsPage.tsx [hunk 6]
```diff
@@ -438,12 +446,22 @@ const LoadedApartmentDetails = ({
...
+    // Handle visibility of the modify person info modal
     const [isModifyPersonInfoModalVisible, setIsModifyPersonInfoModalVisible] = useState(false);
     const [owner, setOwner] = useState<IOwner | undefined>(undefined);
     const modifyPersonInfoHandler = (owner: IOwner) => {
         setIsModifyPersonInfoModalVisible(true);
         setOwner(owner);
     };
+    // Handle visibility of the modify manager info modal
+    const [isModifyManagerModalVisible, setIsModifyManagerModalVisible] = useState(false);
+    const [manager, setManager] = useState<IPropertyManager | undefined>(undefined);
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/features/apartment/ApartmentDetailsPage.tsx]: You are right, it's not a list as I misthought at first :) I changed it.

File: frontend/src/features/apartment/ApartmentDetailsPage.tsx [hunk 7]
```diff
@@ -630,6 +663,11 @@ const LoadedApartmentDetails = ({
                     isVisible={isModifyPersonInfoModalVisible}
                     setIsVisible={setIsModifyPersonInfoModalVisible}
                 />
+                <ModifyManagerModal
+                    manager={manager as IPropertyManager}
+                    isVisible={isModifyManagerModalVisible}
+                    setIsVisible={setIsModifyManagerModalVisible}
+                />
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/features/apartment/ApartmentDetailsPage.tsx]: I added the property manager modification feature to the housing company page too.

File: frontend/src/features/codes/ManagersList.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    }
+    if (params.email?.length && params.email?.length < MIN_LENGTH) {
+        delete params.email;
+    }
+    // get the data
+    const {data, error, isLoading} = useGetPropertyManagersQuery({...params, limit: MAX_ROWS});
+
+    // state for the modal
+    const [isModifyManagerModalOpen, setIsModifyManagerModalOpen] = useState(false);
+    const [manager, setManager] = useState<IPropertyManager | undefined>(undefined);
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/features/codes/ManagersList.tsx]: The component ManagersList does not exist anymore, as there is now a more generic MutateSearchList component instead.

File: frontend/src/features/codes/ManagersList.tsx [hunk 9]
```diff
@@ -50,14 +51,29 @@ const ManagerResultList: React.FC<{params: IFilterManagersQuery}> = ({params}) =
...
+            </div>
             <ModifyManagerModal
                 manager={manager as IPropertyManager}
                 isVisible={isModifyManagerModalOpen}
                 setIsVisible={setIsModifyManagerModalOpen}
             />
+            <ModifyManagerModal
+                isVisible={isAddNewManagerModalOpen}
+                setIsVisible={setIsAddNewManagerModalOpen}
+            />
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/features/codes/ManagersList.tsx]: The ManagersList component does not exist anymore, as there is now a MutateSearchList component instead.

**Review by vincit-matu — APPROVED** [review]
Seems to be working fine, and all the issues Eemeli brought up were addressed. 👍

**Pull request closed without merging**